File: EndGate/EndGate.Core.Client.JS/GameTime.js
var EndGate;
(function (EndGate) {
    (function (Core) {
        var GameTime = (function () {
            function GameTime() {
                this._type = "GameTime";
                this.Now = new Date();
                this._start = this.Now.getTime();
            }
            GameTime.prototype.Update = function () {
                var now = new Date(), nowMs = now.getTime();
                this.Elapsed = nowMs - this.Now.getTime();
                this.Total = nowMs - this._start;
                this.Now = now;
            };
            return GameTime;
        })();
        Core.GameTime = GameTime;        
    })(EndGate.Core || (EndGate.Core = {}));
    var Core = EndGate.Core;
})(EndGate || (EndGate = {}));

File: EndGate/EndGate.Core.Net/GameTime.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EndGate.Core.Net.Interfaces;

namespace EndGate.Core.Net
{
    public class GameTime
    {
        private readonly DateTime _start;

        public GameTime()
        {
            _start = DateTime.UtcNow;
            Now = _start;
        }

        public TimeSpan Elapsed { get; private set; }
        public TimeSpan Total { get; private set; }
        public DateTime Now { get; private set; }

        public void Update()
        {
            var now = DateTime.UtcNow;

            Elapsed = now - Now;
            Total = now - _start;
            Now = now;
        }
    }
}
